src/SeoInspector.js

```diff
@@ -0,0 +1,148 @@
+
+const defaultRules = {
+    'ImgTagWithAltAttritube': { enabled: 1 },
+    'ATagWithRelAttritube': { enabled: 1 },
+    'HeadTagWithTitleAndDescriptionsKeywordsMeta': { enabled: 1 },
+    'NoTooManyStrongTags': { enabled: 1, threshold: 15 },
+    'NoMoreThanOneH1Tag': { enabled: 1 }
+};
+
+class SeoInspector {
+
+    constructor(options = {}) {
+        this.done = (options.done) ? options.done : (err) => { if (err) throw err };
+        this.input = null;
+        this.output = null;
+        this.rules = {};
+        this.ruleFactory = new RuleFactory();
+        return this;
+    }
+
+    read(input) {
+        this.input = input;
+        return this;
+    }
+
+    addRule(name, options) {
+        if (_.has(defaultRules, name)) {
+            _.set(this.rules, name, options);
+        }
+        return this;
+    }
+
+    write(output) {
+        this.output = output;
+        if (_.isEmpty(this.rules)) {
+            this.rules = defaultRules;
+        }
+        this._run();
+    }
+
+    _run() {
+        this._getHtml()
+        .then(
+            html => this._getDom(html)
+        )
+        .then(
+            dom => this._getSummary(dom)
+        )
+        .then(
+            summary => this._output(summary)
+        );
+    }
+
+    _getHtml() {
+    	return new Promise((resolve, reject) => {
+    	    const type = this.input.constructor.name;
+    	    switch (type) {
+    	        case 'String':
+    	            const filePath = this.input;
+    	            fs.readFile(filePath, 'utf8', (err, html) => {
+    	                if (err) return this.done(err);
+    	                resolve(html);
+    	            });
+    	            break;
+    	        case 'ReadStream':
+    	            const readStream = input;
+    	            let chunks = [];
+    	            let size = 0;
+    	            readStream.on('data', (chunk) => {
+    	                chunks.push(chunk);
+    	                size += chunk.length
+    	            });
+    	            readStream.on('end', () => {
+    	                let buffer = Buffer.concat(chunks, size);
+    	                let contentStr = iconv.decode(buffer, 'utf8');
+    	                resolve(contentStr);
+    	            });
+    	            break;
+    	        default:
+    	            return this.done(new Error('read() input is invalid, supported list: a HTML file path and Node Readable Stream.'));
+    	    }
+    	});
+    }
+
+    _getDom(html) {
+        return new Promise((resolve, reject) => {
+            let dom = new JSDOM(html);
+            resolve(dom);
+        });
+    }
+    
+    _getSummary(dom) {
+        return new Promise((resolve, reject) => {
+            let summary = [];
+            _.forEach(this.rules, (options, name) => {
+                let ruleObj = this.ruleFactory.create(name, options, this.done);
+                let report = ruleObj.check(dom)
+                if (report && report != 'undefined') {
```

Comment:
null, undefined and empty string are all 'falsy value'

According to your ruleObject implementation, you can check by this:

`if (!report)`

Reply:
done => if (report) summary.push